### PR TITLE
io_tls: load CA certificates using Crypto API on Windows

### PR DESCRIPTION
**Pull request by fujimotos**
Previously, we tried to load certificates from /etc/ssl/certs if
neither cert_file nor cert_path was specified. This simply does not
work on Windows, since Windows stores certificates in the registry.

This implemnts a Windows-specific routine to load system CA certs
based on Wincrypt.h.

Part of #960

### CODE REVIEW

File: src/flb_io_tls.c [hunk 1]
```diff
@@ -71,6 +72,51 @@ static inline int io_tls_event_switch(struct flb_upstream_conn *u_conn,
...
+#ifdef _MSC_VER
+static int flb_tls_load_system_cert(struct flb_tls_context *ctx)
+{
+    int ret;
+    HANDLE h;
+    PCCERT_CONTEXT cert = NULL;
+
+    h = CertOpenSystemStoreA(NULL, "Root");
+    if (h == NULL) {
+        flb_error("[TLS] Cannot open cert store: %i", GetLastError());
```
Comment (cosmo0920) [inline src/flb_io_tls.c]: Can we obtain and print detail error message with [`FormatMessageA`](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-formatmessagea) here?
(It is not urgent task, just a recommendation for future improving.)

File: src/flb_io_tls.c [hunk 2]
```diff
@@ -71,6 +72,51 @@ static inline int io_tls_event_switch(struct flb_upstream_conn *u_conn,
...
+                                         cert->pbCertEncoded,
+                                         cert->cbCertEncoded);
+            if (ret) {
+                flb_debug("[TLS] cannot parse a certificate. skipping...");
+            }
+        }
+    }
+
+    if (!CertCloseStore(h, 0)) {
+        flb_error("[TLS] Cannot close cert store: %i", GetLastError());
```
Comment (cosmo0920) [inline src/flb_io_tls.c]: ditto.

File: src/flb_io_tls.c (hunk 1 above)
Comment (fujimotos) [reply src/flb_io_tls.c]: Thanks. I'll try to include the logging improvement in later patches.

**Comment by fujimotos** [comment]
Subsumed into the HTTP/ES/Splunk plugin support patch. Closed

**Pull request closed without merging**